formats/json-tests/commonTest/src/kotlinx/serialization/features/sealed/SealedDiamondTest.kt

```diff
@@ -32,8 +41,8 @@ class SealedDiamondTest : JsonTestBase() {
         data class Carrier(val a: A, val b: B, val c: C)
         assertJsonFormAndRestored(
             Carrier.serializer(),
-            Carrier(X(1), X(2), X(3)),
-            """{"a":{"type":"X","i":1},"b":{"type":"X","i":2},"c":{"type":"X","i":3}}"""
+            Carrier(X(1), X(2), Y),
```

Comment:
I found the test change difficult to follow without the fix. Could you please improve the message in `ClassSerialDescriptorBuilder`, e.g.
```
        require(uniqueNames.add(elementName)) { "Element with name '$elementName' is already registered in $serialName" }
```